### PR TITLE
Limit package icon to have only 'png' files

### PR DESCRIPTION
**Pull request by pramodya1994**
- Fixes https://github.com/ballerina-platform/ballerina-lang/issues/34030

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by xlight05 — DISMISSED** [review]

**Review by azinneera — APPROVED** [review]

**Comment by hasithaa** [comment]
Shouldn't we check the file content as well? some can add an SVG file renaming to png file. @ayomawdb, thoughts on this?

**Comment by ayomawdb** [comment]
@hasithaa Agreed. We should ideally validate the magic-bytes of PNG [1][2]

[1] https://en.wikipedia.org/wiki/List_of_file_signatures
[2] https://www.filesignatures.net/index.php?page=search&search=PNG&mode=EXT

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/FileUtils.java [hunk 1]
```diff
@@ -147,7 +152,35 @@ public static void deletePath(Path path) throws IOException {
...
-        
+
+    /**
+     * Check if the given image is a PNG.
+     *
+     * @param filePath image file path
+     * @return is valid PNG file
+     * @throws IOException error when reading the given file path
+     */
+    public static boolean isValidPng(Path filePath) throws IOException {
```
Comment (ayomawdb) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/FileUtils.java:163]: Shall we change this to a generic method usable to validate any file against any provided magic byte. It's fine to have `isValidPng` as well, but ideally we should extract the core logic into a generic reusable method IMO, which accepts filename, hexvalue (magic byte), and offset.

File: compiler/ballerina-lang/src/test/java/io/ballerina/projects/utils/TestFileUtils.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
+/*
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (azinneera) [inline compiler/ballerina-lang/src/test/java/io/ballerina/projects/utils/TestFileUtils.java:2]: Wrong license year

Comment (pramodya1994) [reply compiler/ballerina-lang/src/test/java/io/ballerina/projects/utils/TestFileUtils.java]: Added 2021

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/FileUtils.java (hunk 1 above)
Comment (pramodya1994) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/FileUtils.java:163]: @ayomawdb Can you pls check added `isMatchingImageFormat` method?

Comment (ayomawdb) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/util/FileUtils.java:163]: +1

**Review by ayomawdb — APPROVED** [review]

**Review by azinneera — APPROVED** [review]

**Pull request merged**